volume-providers/vpc/provider/delete_volume.go

```diff
@@ -54,3 +59,25 @@ func validateVolume(volume *provider.Volume) (err error) {
 	err = userError.GetUserError("InvalidVolumeID", nil, volume.VolumeID)
 	return
 }
+
+// WaitForVolumeDeletion checks the volume for valid status
+func WaitForVolumeDeletion(vpcs *VPCSession, volumeID string) (err error) {
+	vpcs.Logger.Debug("Entry of WaitForVolumeDeletion method...")
+	defer vpcs.Logger.Debug("Exit from WaitForVolumeDeletion method...")
+
+	vpcs.Logger.Info("Getting volume details from VPC provider...", zap.Reflect("VolumeID", volumeID))
+
+	var volume *models.Volume
+	err = retry(vpcs.Logger, func() error {
+		vpcs.Logger.Info("Waiting for volume to get deleted.", zap.Reflect("VolumeID", volumeID))
+		volume, err = vpcs.Apiclient.VolumeService().GetVolume(volumeID, vpcs.Logger)
```

Comment:
We should check the real status of volume  from backend ...not just `err!=nil `
Can you use new retry method as used here ---> https://github.com/IBM/ibmcloud-storage-volume-lib/blob/master/volume-providers/vpc/provider/detach_volume.go#L44

Reply:
Once the volume is deleted, we would get error. No way, to check whether it is deleted or wrong volume ID. @pramodgavali

Reply:
I think we need to check the error type i.e `not_found` it means volume got deleted for other errors we have to retry until timeout.